cl/api/utils.py

```diff
@@ -540,3 +545,123 @@ def get_avg_ms_for_endpoint(endpoint: str, d: datetime) -> float:
+    webhook_event: WebhookEvent, response: Response = None, error: str = ""
+) -> None:
+    """Update the webhook event after sending the POST request. If the webhook
+    event fails, increase the retry counter, next retry date and increase its
+    parent webhook failure count. If the webhook event reaches the max retry
+    counter marks it as Failed. If there is no error marks it as Successful.
+
+    :param webhook_event: The WebhookEvent to update.
+    :param response: Optional in case we receive a requests Response object, to
+    update the WebhookEvent accordingly.
+    :param error: Optional, if we don't receive a request Response we would
+    receive an error to log it.
+    :return: None
+    """
+
+    WEBHOOK_MAX_RETRY_COUNTER = 7
+
+    failed_request = False
+    if response is not None:
+        webhook_event.status_code = response.status_code
+        webhook_event.response = response.text
+        # If the response status code is not between 200 and 400, it's
+        # considered a failed attempt and will be enqueued for retry.
+        if not response.ok:
+            failed_request = True
+    if failed_request or error:
+        webhook = webhook_event.webhook
+        webhook.failure_count = F("failure_count") + 1
+        # Avoid notify to admins in cl.users.signals webhook_created_or_updated
+        webhook.save(update_fields=["failure_count"])
+        if webhook_event.retry_counter >= WEBHOOK_MAX_RETRY_COUNTER:
+            # If the webhook has reached the max retry counter, mark as failed
+            webhook_event.event_status = WEBHOOK_EVENT_STATUS.FAILED
+            webhook_event.save()
+            return
+
+        webhook_event.retry_counter = F("retry_counter") + 1
+        webhook_event.save()
+        webhook_event.refresh_from_db()
+        webhook_event.next_retry_date = get_next_webhook_retry_date(
+            webhook_event.retry_counter
+        )
+        webhook_event.event_status = WEBHOOK_EVENT_STATUS.ENQUEUED_RETRY
+        webhook_event.error_message = error
+    else:
+        webhook_event.event_status = WEBHOOK_EVENT_STATUS.SUCCESSFUL
+    webhook_event.save()
+
+
+def send_webhook_event(
+    webhook_event: WebhookEvent, content_str: str = None
+) -> None:
+    """Send the webhook POST request.
+
+    :param webhook_event: An WebhookEvent to send.
+    :param content_str: Optional, the str JSON content to send the first time
+    the webhook is sent.
+    """
+    headers = {
+        "Content-type": "application/json",
+        "Idempotency-Key": str(webhook_event.event_id),
+    }
+    if content_str:
+        json_str = content_str
+    else:
+        renderer = JSONRenderer()
+        json_str = renderer.render(
+            webhook_event.content,
+            accepted_media_type="application/json;",
+        ).decode()
+    try:
+        response = requests.post(
+            webhook_event.webhook.url,
+            data=json_str,
+            timeout=5,
```

Comment:
I think making this just a few seconds is probably better. Clients shouldn't be doing processing when we hit their servers, and if they timeout because their server is busy, well, we'll be back with a retry.

Reply:
Of course, so 1 or 2, seconds would be ok?

Reply:
Changed. I set 1 second for the connection time out and 1 for the read time out.